### PR TITLE
[knex] Update enum arguments to reflect latest signature

### PR DESCRIPTION
**Pull request by willsoto**
See https://knexjs.org/#Schema-enum
https://github.com/tgriesser/knex/pull/2632
https://github.com/tgriesser/knex/issues/394

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://knexjs.org/#Schema-enum
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by mastermatt — COMMENTED** [review]
I'm not sure what's up with Travis, but once that is running you have a 👍 from me.

File: types/knex/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for Knex.js 0.14
+// Type definitions for Knex.js 0.15.2
```
Comment (DanielRosenwasser) [inline types/knex/index.d.ts]: Looks like our lint rules are disabled for this package. Can you enable the lints in our `tslint.json` and fix up the version string to only be a `MAJOR.MINOR` version (e.g. `4.7`), as opposed to including the patch or containing a leading `v` (e.g. `v1.2.3`).

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]
Thanks for your patience! Just try to enable as many lint rules as you can and we can merge things in.

File: types/knex/index.d.ts (hunk 1 above)
Comment (willsoto) [reply types/knex/index.d.ts]: @DanielRosenwasser I am not sure I see where to do that. Could you point me at an example or what I need to tweak?

Comment (DanielRosenwasser) [reply types/knex/index.d.ts]: Just remove the rules field entirely since they override the fields that they extend from `dtslint/dt.json`

Comment (willsoto) [reply types/knex/index.d.ts]: 👍 After enabling the rules, there are dozens of failures. I've pushed up the change anyway.

**Comment by DanielRosenwasser** [comment]
Oof - okay, looking at the build logs this really isn't your problem to fix. Sorry I asked you to turn this on. Just revert that last commit and I can pull this in.

**Comment by willsoto** [comment]
Not a problem. I'll spend some time this weekend working through what is easily fixed and turn off the rest. Thanks!

**Comment by DanielRosenwasser** [comment]
Thank you so much!

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
⭐️ ⭐️ ⭐️ ⭐️ ⭐️ for the patience and going beyond

**Comment by willsoto** [comment]
Happy to help! Thanks for the attentiveness.

**Pull request merged**